xlconnect.Rmd

```diff
@@ -0,0 +1,544 @@
+addImage(wb, 
+         filename = "xlconnect-demo/swissfrancplot.png", 
+         name = "graph", 
+         originalSize = TRUE)
+
+# Save workbook - this actually writes the file
+saveWorkbook(wb)
+```
+
+
+## Other functionalities 
+
+Let's perform some usual tasks on a template and use few more interesting XLConnect functionalities like:
+
+- ensure values get updated when opening a workbook
+
+- check if a sheet is visible
+
+- check if a named region exists
+
+- hide or unhide sheets
+
+- append a named region
+
+- get the coordinates of a named region in a sheet
+
+- additional formatting, e.g. setting the column width
+
+- remove sheets and named region
+
+- clone a sheet
+
+- add a hyperlink
+
+### Recalculate workbook
+Reading....
+```{r , echo = TRUE, eval = FALSE}
+if (!exists("xlconnect-demo/XLCDemo-TplCopy.xlsx")) {
+  # Create a copy of the template
+  file.copy("xlconnect-demo/XLConnectDemo-Tpl.xlsx", 
+          "xlconnect-demo/XLCDemo-TplCopy.xlsx", 
+          overwrite = FALSE)
+} 
+
+# Load copy of the template
+wbcopy <- loadWorkbook("xlconnect-demo/XLCDemo-TplCopy.xlsx")
+
+# Read data from template copy using named regions
+inc_df <- readNamedRegion(wbcopy,
+                          name = "Income_inp",
+                          header = TRUE)
+persexp_df <- readNamedRegion(wbcopy,
+                          name = "PersExpenses_inp",
+                          header = TRUE)
+opexp_df <- readNamedRegion(wbcopy,
+                          name = "OpExpenses_inp",
+                          header = TRUE)
+
+# Update numbers: double sales
+inc_df$ACTUAL[inc_df$INCOME == "Net sales"] <- inc_df$ACTUAL[inc_df$INCOME == "Net sales"] * 2
+# Update numbers: add employee benefits
+persexp_df$ACTUAL[persexp_df$PERSONNEL.EXPENSES == "Employee benefits"] <- 3000
+# Update numbers: reduce maintenance
+opexp_df$ACTUAL[opexp_df$OPERATING.EXPENSES == "Maintenance and repairs"] <- 2500
+```
+
+Writing...
+```{r , echo = TRUE, eval = FALSE}
+# Update output accordingly
+writeNamedRegion(wbcopy, Sys.Date(), "Date", header = FALSE)
+writeNamedRegion(wbcopy, inc_df, "Income_out", header = FALSE)
+writeNamedRegion(wbcopy, persexp_df, "PersExpenses_out", header = FALSE)
+writeNamedRegion(wbcopy, opexp_df, "OpExpenses_out", header = FALSE)
+
+# Get numbers of sheets
+nb_sheets <- length(getSheets(wbcopy))
+
+# Force Excel to re-calculate when opening the workbook
+setForceFormulaRecalculation(wbcopy, sheet = seq(1, nb_sheets), TRUE)
+
+# Save workbook - this actually writes the file
+saveWorkbook(wbcopy)
+```
+
+### Hide sheets
+Hide irrelevant sheets
+```{r , echo = TRUE, eval = FALSE}
+# Hide sheets
+sheetsToHide <- c("Style", "Input")
+
+# Ensure those sheets exist in the workbook
+sheetsToHide <- sheetsToHide[sheetsToHide %in% getSheets(wbcopy)]
+hideSheet(wbcopy, sheetsToHide, veryHidden = TRUE)
+
+# Save workbook - this actually writes the file
+saveWorkbook(wbcopy)
+```
+
+### Sheet visibility check and unhide sheets 
```

Comment:
```suggestion
### Sheet visibility check and unhide sheets
Check if a given sheet is visible and unhide it if it is.
```
Consistency: Otherwise all the points listed with ## / ### have an introductory sentence